### PR TITLE
CASMINST-4085: change the external-dns image path

### PR DESCRIPTION
**Pull request by bo-quan**
## Summary and Scope

External-dns image path is not consistent with the pattern we use for other container images. This fixes the image path for the externa-dns image.

## Issues and Related PRs

* Resolves [CASMINST-4085]

### Tested on:

  * Virtual Shasta

### Test description:

This does not change the image itself, and will be tested with new chart that deploys the image.

- Were the install/upgrade-based validation checks/tests run (goss tests/install-validation doc)?
- Were continuous integration tests run? If not, why?
- Was upgrade tested? If not, why?
- Was downgrade tested? If not, why?
- Were new tests (or test issues/Jiras) created for this change?

## Risks and Mitigations

Low. This won't change the image - only the path reference that will be tested with the new chart and manifest.

## Pull Request Checklist

- [ ] Version number(s) incremented, if applicable
- [x] Copyrights updated
- [x] License file intact
- [x] Target branch correct
- [ ] CHANGELOG.md updated
- [ ] Testing is appropriate and complete, if applicable
- [ ] [HPC Product Announcement](https://cray.slack.com/archives/C026TVCSXLH) prepared, if applicable

### CODE REVIEW

File: .github/workflows/docker.io.bitnami.external-dns.0.10.2-debian-10-r23.yaml [hunk 1]
```diff
@@ -28,22 +28,22 @@ on:
     branches:
       - main
     paths:
-      - .github/workflows/docker.io.bitnami.external-dns.0.debian-10.yaml
-      - docker.io/bitnami/external-dns/0/debian-10/**
+      - .github/workflows/docker.io.bitnami.external-dns.0.10.2-debian-10-r23.yaml
+      - docker.io/bitnami/external-dns/**
```
Comment (mtupitsyn) [inline .github/workflows/docker.io.bitnami.external-dns.0.10.2-debian-10-r23.yaml:32]: I think this should be

    - docker.io/bitnami/external-dns/0.10.2-debian-10-r23/**
 
The idea is that it needs to be re-triggered, when any of the file for specific image tag is changing.

File: .github/workflows/docker.io.bitnami.external-dns.0.10.2-debian-10-r23.yaml [hunk 2]
```diff
@@ -28,22 +28,22 @@ on:
...
   pull_request:
     branches:
       - main
     paths:
-      - .github/workflows/docker.io.bitnami.external-dns.0.debian-10.yaml
-      - docker.io/bitnami/external-dns/0/debian-10/**
+      - .github/workflows/docker.io.bitnami.external-dns.0.10.2-debian-10-r23.yaml
+      - docker.io/bitnami/external-dns/**
```
Comment (mtupitsyn) [inline .github/workflows/docker.io.bitnami.external-dns.0.10.2-debian-10-r23.yaml:38]: Same here

File: .github/workflows/docker.io.bitnami.external-dns.0.10.2-debian-10-r23.yaml [hunk 3]
```diff
@@ -28,22 +28,22 @@ on:
...
   workflow_dispatch:
 
 jobs:
   build:
     runs-on: self-hosted
     env:
-      CONTEXT_PATH: docker.io/bitnami/external-dns/0/debian-10
-      DOCKER_REPO: artifactory.algol60.net/csm-docker/stable/docker.io/bitnami/external-dns/0/debian-10
+      CONTEXT_PATH: docker.io/bitnami/external-dns
```
Comment (mtupitsyn) [inline .github/workflows/docker.io.bitnami.external-dns.0.10.2-debian-10-r23.yaml:45]: Same here, CONTEXT_PATH should include a tag

File: .github/workflows/docker.io.bitnami.external-dns.0.10.2-debian-10-r23.yaml (hunk 1 above)
Comment (bo-quan) [reply .github/workflows/docker.io.bitnami.external-dns.0.10.2-debian-10-r23.yaml]: fixed

File: .github/workflows/docker.io.bitnami.external-dns.0.10.2-debian-10-r23.yaml (hunk 3 above)
Comment (bo-quan) [reply .github/workflows/docker.io.bitnami.external-dns.0.10.2-debian-10-r23.yaml]: fixed

File: .github/workflows/docker.io.bitnami.external-dns.0.10.2-debian-10-r23.yaml (hunk 2 above)
Comment (bo-quan) [reply .github/workflows/docker.io.bitnami.external-dns.0.10.2-debian-10-r23.yaml]: fixed

**Review by mtupitsyn — APPROVED** [review]
Looks good!

**Pull request merged**